### PR TITLE
JNG-5543 The Optional is null when the first letter of an attribute is uppercase

### PR DESCRIPTION
**Pull request by gaborflorian**
<table>
<td>
  <a href="https://blackbelt.atlassian.net/browse/JNG-5543" title="JNG-5543" target="_blank">BugJNG-5543</a>  If the first letter of an attribute name is uppercase in JSL the Optional is null in Java
  </td></table>
  <br />
 

JNG-5543 The Optional is null when the first letter of an attribute is uppercase

### CODE REVIEW

File: judo-runtime-core-jsl-itest/models/SpecialCasesModel/src/test/java/hu/blackbelt/judo/runtime/core/jsl/entity/SpecialCasesTest.java [hunk 1]
```diff
@@ -2149,4 +2168,209 @@ void testSafeFieldNames() {
...
+            "REQ-ENT-007",
+            "REQ-MDL-001",
+            "REQ-MDL-002",
+            "REQ-MDL-003",
+    })
+    void testUpperFieldNames() {
+
+        PrimUpper primUpper = primUpperDao.create(PrimUpperForCreate.builder()
+                .withAbc("Abc")
+                .withABd("ABc")
```
Comment (noherczeg) [inline judo-runtime-core-jsl-itest/models/SpecialCasesModel/src/test/java/hu/blackbelt/judo/runtime/core/jsl/entity/SpecialCasesTest.java]: This is misleading, test data should mirror the attribute IMO.

**Review by noherczeg — CHANGES_REQUESTED** [review]

File: judo-runtime-core-jsl-itest/models/SpecialCasesModel/src/test/resources/SpecialCases.jsl [hunk 2]
```diff
@@ -122,4 +122,44 @@ entity Case {
 }
 
 entity Static {
-}
\ No newline at end of file
+}
+
+entity PrimUpper {
+    field String Abc;
```
Comment (noherczeg) [inline judo-runtime-core-jsl-itest/models/SpecialCasesModel/src/test/resources/SpecialCases.jsl:128]: If supported by JSL, please add another ref/field, etc with the name of "abc". If we are case sensitive, we need to make sure that different fields with different casings can have dedicated values.

**Review by noherczeg — CHANGES_REQUESTED** [review]

File: judo-runtime-core-jsl-itest/models/SpecialCasesModel/src/test/resources/SpecialCases.jsl (hunk 2 above)
Comment (gaborflorian) [reply judo-runtime-core-jsl-itest/models/SpecialCasesModel/src/test/resources/SpecialCases.jsl:128]: Not supported: using the same field name when the only difference is the capitalization.

File: judo-runtime-core-jsl-itest/models/SpecialCasesModel/src/test/java/hu/blackbelt/judo/runtime/core/jsl/entity/SpecialCasesTest.java (hunk 1 above)
Comment (gaborflorian) [reply judo-runtime-core-jsl-itest/models/SpecialCasesModel/src/test/java/hu/blackbelt/judo/runtime/core/jsl/entity/SpecialCasesTest.java]: [c88c8c2](https://github.com/BlackBeltTechnology/judo-runtime-core-jsl/pull/294/commits/c88c8c230544f4be6a02b3157afcc721923416a5)

File: judo-runtime-core-jsl-itest/models/SpecialCasesModel/src/test/resources/SpecialCases.jsl (hunk 2 above)
Comment (gaborflorian) [reply judo-runtime-core-jsl-itest/models/SpecialCasesModel/src/test/resources/SpecialCases.jsl:128]: Screenshot 2024-03-12 at 12 59 37

**Review by noherczeg — APPROVED** [review]

**Review by Madmat8 — APPROVED** [review]

**Review by robertcsakany — APPROVED** [review]

**Pull request merged**